src/flows/Termination/tests/TerminationFlow.test.tsx

```diff
@@ -0,0 +1,491 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import React from 'react';
+import { PropsWithChildren } from 'react';
+import { beforeEach, describe, it, vi } from 'vitest';
+import { server } from '@/src/tests/server';
+import {
+  RenderProps,
+  TerminationFlow,
+} from '@/src/flows/Termination/TerminationFlow';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {
+  fillCheckbox,
+  fillRadio,
+  fillSelect,
+  selectDayInCalendar,
+} from '@/src/common/testHelpers';
+import { http, HttpResponse } from 'msw';
+import { terminationResponse } from '@/src/flows/Termination/tests/fixtures';
+
+const queryClient = new QueryClient();
+
+const wrapper = ({ children }: PropsWithChildren) => (
+  <QueryClientProvider client={queryClient}>
+    <FormFieldsProvider components={{}}>{children}</FormFieldsProvider>
+  </QueryClientProvider>
+);
+
+const mockOnSubmit = vi.fn();
+const mockOnSuccess = vi.fn();
+const mockOnError = vi.fn();
+
+describe('TerminationFlow', () => {
+  const mockRender = vi.fn(({ terminationBag, components }: RenderProps) => {
+    const { Form, Back, SubmitButton } = components;
+
+    const currentStepIndex = terminationBag.stepState.currentStep.index;
+
+    const steps: Record<number, string> = {
+      [0]: 'Employee Communication',
+      [1]: 'Termination Details',
+      [2]: 'Paid Time Off',
+      [3]: 'Additional Information',
+    };
+
+    return (
+      <>
+        <h1>Step: {steps[currentStepIndex]}</h1>
+        <Form
+          username="ze"
+          onSubmit={mockOnSubmit}
+          onSuccess={mockOnSuccess}
+          onError={mockOnError}
+        />
+        {currentStepIndex > 0 && <Back>Back</Back>}
+        {currentStepIndex <= terminationBag.stepState.totalSteps - 1 && (
+          <SubmitButton>
+            {currentStepIndex < terminationBag.stepState.totalSteps - 1
+              ? 'Next Step'
+              : 'Send termination'}
+          </SubmitButton>
+        )}
+      </>
+    );
+  });
+  const defaultProps = {
+    employmentId: '2ef4068b-11c7-4942-bb3c-70606c83688e',
+    countryCode: 'PRT',
+    options: {},
+    render: mockRender,
+  };
+
+  let offboardingRequest: Record<string, unknown> | null = null;
+  beforeEach(() => {
+    vi.clearAllMocks();
+
+    server.use(
+      http.post('*/v1/offboardings', async (req) => {
+        offboardingRequest = (await req.request.json()) as Record<
+          string,
+          unknown
+        >;
+        return HttpResponse.json(terminationResponse);
+      }),
+    );
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  function getYearMonthDate(date: Date) {
+    const year = date.getFullYear();
+    const month = String(date.getMonth() + 1).padStart(2, '0');
+    const day = date.getDay();
+
+    return {
+      year,
+      month,
+      day,
+    };
+  }
+
+  async function fillStep1(
```

Comment:
nit: I would rename this function to `fillEmployeeCommunication`. I think it's more explicit! Same goes to the other functions bellow